lib/spack/spack/package.py

```diff
@@ -782,11 +790,25 @@ def stage(self, stage):
 
     @property
     def env_path(self):
-        return os.path.join(self.stage.path, 'spack-build.env')
+        """The Spack build environment file path."""
+        return os.path.join(self.stage.path, _spack_build_envfile)
+
+    def install_env_path(self, relative=False):
+        """Return the optionally relative path to the install env file."""
+        spec = self.spec if not relative else None
+        return os.path.join(spack.store.layout.metadata_path(spec),
+                            _spack_build_envfile)
 
     @property
     def log_path(self):
-        return os.path.join(self.stage.path, 'spack-build.txt')
+        """The Spack build log file path."""
+        return os.path.join(self.stage.path, _spack_build_logfile)
+
+    def install_log_path(self, relative=False):
+        """Return the optionally relative path to the install log file."""
+        spec = self.spec if not relative else None
+        return os.path.join(spack.store.layout.metadata_path(spec),
```

Comment:
I was under the impression from @tgamblin's post (https://github.com/spack/spack/pull/11806#discussion_r297864037) and possibly off-line discussion that the file paths should not be in `directory_layout`.

Did I misunderstand?

Reply:
FYI. I am awaiting feedback on whether the install paths should be moved back to `directory_layout.py` before making the changes.

Reply:
Having the derived metadata directories in `Package` seems fine, although I think that means `build_packages_path` should also be in `Package`. I'm not sure what

> just move these things under metadata_path

means because it appears that was already the case. If the idea is to maintain the same directory structure between the stage and install directories, I think that is less important than maintaining the same filenames (but it doesn't seem problematic to keep `.stage/spack-build-log.txt` in the stage either).

Reply:
IIRC I thought the same thing about `build_packages_path` but Todd (@tgamblin) didn't want that change.  My impression was that Todd wanted to distinguish between directories (in `directory_layout.py`) and generated/standard filenames (e.g., `log` and `env` paths).